core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/MultiCloudWorkspaceService.scala

```diff
@@ -233,33 +226,19 @@ class MultiCloudWorkspaceService(override val ctx: RawlsRequestContext,
 
     val wsmConfig = multiCloudWorkspaceConfig.workspaceManager
       .getOrElse(throw new RawlsException("WSM app config not present"))
-
-    def createNewWorkspaceRecord(): Future[Workspace] =
-      dataSource.inTransaction { access =>
-        for {
-          _ <- failIfWorkspaceExists(request.toWorkspaceName)
-          workspaceId = UUID.randomUUID()
-          newWorkspace <- createMultiCloudWorkspaceInDatabase(
-            workspaceId.toString,
-            request.toWorkspaceName,
-            request.attributes,
-            access,
-            parentContext
-          )
-        } yield newWorkspace
-      }
+    val workspaceId = UUID.randomUUID()
 
     for {
       // The call to WSM is asynchronous. Before we fire it off, allocate a new workspace record
       // to avoid naming conflicts - we'll erase it should the clone request to WSM fail.
-      newWorkspace <- createNewWorkspaceRecord()
+      newWorkspace <- createNewWorkspaceRecord(workspaceId, request, parentContext)
```

Comment:
It would be nice to connect the provisional creation with the deletion on failure, in a more direct way, with some consistency across instances. Might be more trouble than it's worth, though.

Reply:
Not entirely sure I understand. Do you mean you'd like to see the rollback logic directly after each WSM call when creating or cloning a workspace instead of just the one catchall I've added for workspace creation?

Reply:
Looked into how `cloneWorkspace` does its rollback, it does clean up the Rawls record if any WSM calls fail but it doesn't delete the workspace in WSM. I think it's fine as is so I'm going to leave it rather than expand the scope of this ticket